controllers/activemqartemis_reconciler.go

```diff
@@ -3056,3 +3161,62 @@ func getDeploymentSize(cr *brokerv1beta1.ActiveMQArtemis) int32 {
 	}
 	return *cr.Spec.DeploymentPlan.Size
 }
+
+// The cert should have keystores configured(jks/pkcs12)
+func (reconciler *ActiveMQArtemisReconcilerImpl) MakeSecretFromCertTlsSecret(cert *cm.Certificate, targetSecretName string, targetSecretNamespace string, client rtclient.Client) (*corev1.Secret, error) {
+	var log = ctrl.Log.WithValues("cert", cert.Name, "ns", cert.Namespace)
+	tlsSecret := corev1.Secret{}
+	tlsSecretKey := types.NamespacedName{Name: cert.Spec.SecretName, Namespace: cert.Namespace}
+
+	log.V(1).Info("retrieving tls secret", "secret key", tlsSecretKey)
+	if err := resources.Retrieve(tlsSecretKey, client, &tlsSecret); err != nil {
+		return nil, err
+	}
+
+	dataMap := make(map[string][]byte)
+	targetSecretNamespacedName := types.NamespacedName{Name: targetSecretName, Namespace: targetSecretNamespace}
+
+	// The secret contents depends on what's in the Certificate's spec
+	// the secret will always have 3 entries: ca.crt, tls.crt and tls.key
+	// PCKS12: the secret will have keystore.p12 entry and truststore.p12 entry
+	// JKS: the secret will have keystore.jks entry and truststore.jks entry
+	if cert.Spec.Keystores != nil {
+		if cert.Spec.Keystores.PKCS12 != nil && cert.Spec.Keystores.PKCS12.Create {
+			// if PKCS12 is specified we use it
+			dataMap["broker.ks"] = tlsSecret.Data["keystore.p12"]
+			keyPass, err := certutil.GetPKCS12StorePasswordFromCert(cert, client)
+			if err != nil {
+				log.Error(err, "fail to retrieve keystore password", "secret namespace", cert.Namespace)
+				return nil, err
+			}
+			dataMap["keyStorePassword"] = keyPass
+			dataMap["client.ts"] = tlsSecret.Data["truststore.p12"]
+			dataMap["trustStorePassword"] = keyPass
+		} else if cert.Spec.Keystores.JKS != nil && cert.Spec.Keystores.JKS.Create {
+			// else if JKS is specified we use it
+			dataMap["broker.ks"] = tlsSecret.Data["keystore.jks"]
+			keyPass, err := certutil.GetJKSStorePasswordFromCert(cert, client)
+			if err != nil {
+				log.Error(err, "fail to retrieve keystore password", "secret namespace", cert.Namespace)
+				return nil, err
+			}
+			dataMap["keyStorePassword"] = keyPass
+			dataMap["client.ts"] = tlsSecret.Data["truststore.jks"]
+			dataMap["trustStorePassword"] = keyPass
+		}
```

Comment:
The reconciler doesn't need to duplicate the certificates, it could point to the certificate of the tlsSecret.

Reply:
The tls secret could be in some other namespace. We copy it to make sure it's still compatible with old locations.

Reply:
Yes, the tls secret must be in the same namespace but in this way the operator will automatically handle the certificate renewal. We could create a secret with [keyStorePath](https://github.com/artemiscloud/activemq-artemis-operator/blob/v1.0.15/controllers/activemqartemis_reconciler.go#L926C61) to point to be compatible.

Reply:
The problem is that how you can mount a secret from a different namespace?

Reply:
This limitation could be a requirement for users that want to use the cert-manager: the secret created by the cert-manger has to be in the same namespace of the ActiveMQArtemis CR.

Reply:
Sorry, if im missing something here.

Can this not work in the same mannger that cert-manage does when dealing with Ingress TLS?

```
ingress:
  enabled: true
  className: "nginx-internal"
  annotations:
    nginx.ingress.kubernetes.io/service-upstream: "true"
    cert-manager.io/issuer: acmpa
    cert-manager.io/issuer-kind: AWSPCAClusterIssuer
    cert-manager.io/issuer-group: awspca.cert-manager.io
    cert-manager.io/common-name: whatever.com
```

Where cert manager see the annotation, and creates the TLS Certificate in the same namespace?

Reply:
@jseiser Hi, there is another task to track the ingress/route support for cert-manager
https://github.com/artemiscloud/activemq-artemis-operator/issues/679